gensim/models/keyedvectors.py

```diff
@@ -90,11 +109,10 @@ def most_similar(self, positive=[], negative=[], topn=10, restrict_vocab=None, i
         for word, weight in positive + negative:
             if isinstance(word, ndarray):
                 mean.append(weight * word)
-            elif word in self.vocab:
-                mean.append(weight * self.syn0norm[self.vocab[word].index])
-                all_words.add(self.vocab[word].index)
             else:
-                raise KeyError("word '%s' not in vocabulary" % word)
+                mean.append(weight * self.word_vec(word, use_norm=True))
+                if word in self.vocab:
```

Comment:
That line raises a `KeyError` in case `word in self.vocab` is `False`. So in case it's `True`, line 115 would be executed.
Also, `word_vec` has been overriden in the `KeyedVectors` subclass for `FastText`.